packages/mobile/src/services/ReactNativeLogger.ts

```diff
@@ -34,13 +41,38 @@ export default class ReactNativeLogger {
     const sanitizedError =
       error && shouldSanitizeError ? this.sanitizeError(error, valueToPurge) : error
     const errorMsg = this.getErrorMessage(sanitizedError)
-    Sentry.captureException(error, { extra: { tag, message, errorMsg, source: 'Logger.error' } })
-    console.info(`${tag} :: ${message} :: ${errorMsg}`)
+    const isNetworkError = this.networkErrors.some((networkError) =>
+      errorMsg.toLowerCase().includes(networkError)
+    )
+
+    // prevent genuine network errors from being sent to Sentry
+    if (!isNetworkError || (this.isNetworkConnected && isNetworkError)) {
+      Sentry.captureException(error, {
+        extra: {
+          tag,
+          message,
+          errorMsg,
+          source: 'Logger.error',
+          networkConnected: this.isNetworkConnected,
+        },
+      })
+    }
+    console.info(
+      `${tag} :: ${message} :: ${errorMsg} :: network connected ${this.isNetworkConnected}`
+    )
     if (__DEV__) {
       console.info(console.trace())
     }
   }
 
+  setIsNetworkConnected = (isConnected: boolean) => {
+    this.isNetworkConnected = isConnected
+  }
+
+  setNetworkErrors = (errors: string[]) => {
+    this.networkErrors = [...this.networkErrors, ...errors]
```

Comment:
Oh sorry one small thing, this has the potential of growing indefinitely everytime the feature flags are fetched.
Would be be better to have a `DEFAULT_NETWORK_ERRORS` array to avoid this.

Reply:
oh yeah you're right, i overlooked this. i added a 'DEFAULT_SENTRY_NETWORK_ERRORS' to the global config.ts file so that it can live with the other sentry config variable, hope this is okay